### DISCUSSION TITLE
argo workflows sso without argocd?

### DISCUSSION THREAD

**Discussion by Hunter-Thompson** [argoproj/argo-workflows#8747] [Q&A]
Id like to setup Github SSO for my argo workflows, but without an argocd installation.

Is this possible?

Im guessing Il have to deploy a dex server and setup Github OAuth. Is there any documentation on how to do this?

**Comment by caelan-io** [comment]
I have experience doing this with GitHub. [This guide](https://medium.com/preply-engineering/k8s-auth-a81f59d4dff6) was helpful for setting up dex on K8s for GitHub.

Once you do that, use this [list of different connectors](https://dexidp.io/docs/connectors/). The workflow-controller-configmap can be configured to directly connect to these connectors without installing Dex.

Finally, set auth mode to SSO on Argo. Then you should be set.

If you'd like more tips beyond that, DM me on the CNCF Slack or email me (c [at] pipekit.io), and we can find a time.
